poetry/core/version/pep440/segments.py

```diff
@@ -45,15 +45,11 @@ def __post_init__(self):
         elif not isinstance(self.extra, tuple):
             object.__setattr__(self, "extra", (self.extra,))
 
-        parts = list(
-            map(
-                str,
-                filter(
-                    lambda x: x is not None,
-                    [self.major, self.minor, self.patch, *self.extra],
-                ),
-            )
-        )
+        parts = [
```

Comment:
The only thing I don't like here is the use of `_` as a variable name when the value is not discarded. This goes against general Python convention. `part` or something similar would be suitable and make it clear that the value is used.